moveit_planners/pilz_industrial_motion_planner/src/joint_limits_aggregator.cpp

```diff
@@ -39,22 +39,33 @@
 #include <moveit/robot_model_loader/robot_model_loader.h>
 
 #include <vector>
+namespace
+{
+static const rclcpp::Logger LOGGER =
+    rclcpp::get_logger("moveit.pilz_industrial_motion_planner.joint_limits_aggregator");
+}
 
 pilz_industrial_motion_planner::JointLimitsContainer
 pilz_industrial_motion_planner::JointLimitsAggregator::getAggregatedLimits(
-    const ros::NodeHandle& nh, const std::vector<const moveit::core::JointModel*>& joint_models)
+    const rclcpp::Node::SharedPtr& node, const std::string& param_namespace,
+    const std::vector<const moveit::core::JointModel*>& joint_models)
 {
   JointLimitsContainer container;
 
-  ROS_INFO_STREAM("Reading limits from namespace " << nh.getNamespace());
+  RCLCPP_INFO_STREAM(LOGGER, "Reading limits from namespace " << param_namespace);
 
   // Iterate over all joint models and generate the map
   for (auto joint_model : joint_models)
   {
     JointLimit joint_limit;
 
-    // If there is something defined for the joint on the parameter server
-    if (joint_limits_interface::getJointLimits(joint_model->getName(), nh, joint_limit))
+    // NOTE: declareParameters fails (=returns false) if the parameters have already been declared.
+    // The function should be checked in the if condition below when we disable
+    // 'NodeOptions::automatically_declare_parameters_from_overrides(true)'
+    joint_limits_interface::declareParameters(joint_model->getName(), param_namespace, node);
```

Comment:
We should make a decision if we want to enforce declaring parameters everywhere or if we are fine with allowing automatic overrides moving forward.